### PR TITLE
response/result.rs: Replace generate_deserialize_metadata macro with a function

### PR DESCRIPTION
**Pull request by Lorak-mmk**
This macro is unnecessary, function is enough here. For ease of future development, lets get rid of such macros.

## Pre-review checklist

- [ ] I have split my patch into logically separate commits.
- [ ] All commit messages clearly explain what they change and why.
- [ ] I added relevant tests for new features and bug fixes.
- [ ] All commits compile, pass static checks and pass test.
- [ ] PR description sums up the changes and reasons why they should be introduced.
- [ ] I have provided docstrings for the public items that I want to introduce.
- [ ] I have adjusted the documentation in `./docs/source/`.
- [ ] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

**Comment by Lorak-mmk** [comment]
v2: Moved `deserialize_metadata_generic` to `impl RawMetadataAndRawRows` and made it a private method.

File: scylla-cql/src/frame/response/result.rs [hunk 1]
```diff
@@ -1099,100 +1099,110 @@ impl RawMetadataAndRawRows {
...
-                            .transpose()
-                            .map_err(ResultMetadataParseError::from)?;
-
-                        let col_specs = $deser_col_specs(
-                            frame_slice.as_slice_mut(),
-                            global_table_spec,
-                            self.col_count,
-                        )
+impl RawMetadataAndRawRows {
+    /// Deserializes ResultMetadata in the form mentioned by its name,
```
Comment (wprzytula) [inline scylla-cql/src/frame/response/result.rs:1116]: No longer true - the name does not mention the form.

File: scylla-cql/src/frame/response/result.rs [hunk 2]
```diff
@@ -1099,100 +1099,110 @@ impl RawMetadataAndRawRows {
...
+        ) -> ResultMetadataHolder<'result>,
+        deser_col_specs: DeserColSpecsFn<'this, 'result>,
+    ) -> StdResult<DeserializedMetadataAndRawRows<'result, T>, RowsParseError> {
+        let mut frame_slice = FrameSlice::new(&self.raw_metadata_and_rows);
+
+        let metadata = match self.cached_metadata.as_ref() {
+            Some(cached) if self.no_metadata => {
+                // Server sent no metadata, but we have metadata cached. This means that we asked the server
+                // not to send metadata in the response as an optimization. We use cached metadata instead.
+                //$use_cached_metadata(cached)
```
Comment (wprzytula) [inline scylla-cql/src/frame/response/result.rs:1135]: leftover commented out code

**Review by wprzytula — CHANGES_REQUESTED** [review]
Huge thanks for this!

**Comment by Lorak-mmk** [comment]
Rebased

**Pull request merged**